### ISSUE TITLE
[BUG] Favorites screen widgets doesn't immediately reflect user's signed in status

### ISSUE THREAD

**Issue by sandeep-ps** [rokwire/illinois-app#2886]
**Describe the bug**
After the user signs in from the Settings screen, some of the Favorites screen widgets don't immediately reflect the signed-in status.

**To Reproduce**
1. Sign out from the app or start when the user is already signed out.
2. Go to the Settings screen and sign in using NetID and password.
3. Close the Settings screen and see that on the home page, "Connect to Illinois" still shows as signed out.
4. During some initial tests, the Connect to Illinois widget showed up for a long time, even after refreshing the Favorites screen by dragging it down.
5. After a couple of minutes, the "Connect to Illinois" widget goes away on the Favorites screen, and the Polls widget shows as signed in.

**Expected behavior**
Once the user signs in, it should be reflected in all widgets immediately.

**Screenshots**
Screenshots of Favorites screen widgets showing signed-out status after the user has signed in from the Settings screen:

![image](https://user-images.githubusercontent.com/4306604/210410901-ef7429f8-07dd-4fb6-ac86-bdfa4ba35c88.png)
![image](https://user-images.githubusercontent.com/4306604/210410856-03d64f03-823d-4338-9ae4-098969e039b5.png)

**Smartphone (please complete the following information):**
 - Device: iPhone SE
 - OS: iOS
 - Version: 4.2.47

**Additional context**

**Comment by mihail-varbanov** [comment]
Hi @sandeep-ps and @pmarkhennessy,

I was unable to recreate this.

In general, we should wait for the login process to complete, in order to expect the UI to reflect on the new login status. There is a progress indicator in the Sign in buttons in Settings panel that is visible while the login process is in progress. Please note that getting back from the Shibboleth browser does not mark the end of the login process, it indicates rather the beginning.

So my suspicion is that some of the HTTP requests that the client app does on login did delay and the Settings panel was closed before the login sequence had finished (indicated by the progress indicator in the Sign in button). The client behavior is consistent if this was the case, and I do not see what reasonable action we can do in this case. Preventing the user from closing the Settings panel until the login process completes is not a good UX for me.

**Comment by sandeep-ps** [comment]
Hi, @mihail-varbanov, 

Thanks. I'm not seeing a progress bar on the Sign In button. Are you referring to the refresh icon (circular arrow) that rotates when the sign-in process is in progress? If so, I saw that, and I closed the Settings screen only after that icon was gone indicating to me that the sign-in process was complete. Let me share one more screenshot that shows the end result on the Favorites screen. At the top, it shows that I'm signed in as it displays my name. Below, you can see the widget indicating that the user needs to sign in. Could you please give it one more try? If you still cannot reproduce the issue, please let me know, and I will close this accordingly. I understand that this may be caused due to some delay, which may not be easily reproducible. Thanks again.

![image](https://user-images.githubusercontent.com/4306604/211349932-efb3e560-acf7-4b15-9744-9e911db86b5c.png)

**Comment by mihail-varbanov** [comment]
Hi @sandeep-ps,

#3111 described the same problem and fortunately provided a scenario that I was able to recreate. The problem should be resolved in version 4.3.46.

**Comment by sandeep-ps** [comment]
Thanks. I will close this.

Duplicate of #3111.

**Comment by sandeep-ps** [comment]
Duplicate of #3111
